### PR TITLE
[CICD-8120] Sync with sgleske upstream

### PR DESCRIPTION
**Pull request by jnathan-ias**
CICD-8120 Sync integralads / jenkins-bootstrap-shared with samrocketman/jenkins-bootstrap-shared

Summary
------

Merge upstream changes from https://github.com/samrocketman/jenkins-bootstrap-shared into Org repo.
Complete implementation of jenkins autoconfig scripts 

See Also 
------
- [jira](https://jira.integralads.com/browse/CICD-8120)
- https://github.com/integralads/jenkins-bootstrap-shared/pull/5
- https://github.com/integralads/jenkins-bootstrap-shared/pull/4
- https://github.com/integralads/jenkins-bootstrap-shared/pull/3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@
 /my_jenkins_home
 /plugins
 /ssh_config
+/venv
```
Comment (bsanchez-ias) [inline .gitignore]: mind adding a empty line

File: Dockerfile [hunk 2]
```diff
@@ -1,21 +1,24 @@
-#https://github.com/anapsix/docker-alpine-java
-FROM nexus.303net.net:8443/anapsix/alpine-java:8_jdk
+#https://hub.docker.com/layers/library/alpine/3.17.3/images/sha256-b6ca290b6b4cdcca5b3db3ffa338ee0285c11744b4a6abaa9627746ee3291d8d?context=explore
+ARG base=nexus.303net.net:8443/alpine:3.17.3
```
Comment (bsanchez-ias) [inline Dockerfile:2]: This has been tested and works as expected. Results can be found [here](https://jira.integralads.com/browse/CICD-8118?focusedCommentId=1099634&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1099634)

File: gradle.properties [hunk 3]
```diff
@@ -1,2 +1,2 @@
 bootstrapHome=.
-version=2.222.1.1
+version=2.361.2.1
```
Comment (bsanchez-ias) [inline gradle.properties]: Please update this to `2.387.2.1`

**Review by bsanchez-ias — APPROVED** [review]
Once `gradle.properties` file has been update this PR can be merged. Jenkins-bootrap-shared can be synced up to latest master branch of https://github.com/samrocketman/jenkins-bootstrap-shared with some custom changes.

File: gradle.properties (hunk 3 above)
Comment (jnathan-ias) [reply gradle.properties]: pushed

**Pull request merged**